kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -206,6 +209,38 @@ object KinesisSink {
       case None => None.asRight
     }
 
+  /**
+    * Check whether an SQS queue (buffer) exists
+    * @param name Name of the queue
+    * @return Whether the queue exists
+    */
+  private def sqsQueueExists(client: AmazonSQS, name: String): Boolean =
+    try {
+      client.getQueueUrl(name)
+      true
+    } catch {
+      case _: QueueDoesNotExistException => false
+    }
+
+  private def runChecks(
+    kinesisClient: AmazonKinesis,
+    streamName: String,
+    sqs: Option[SqsClientAndName]
+  ): Unit = {
+    lazy val log = LoggerFactory.getLogger(getClass())
+
+    // format: off
+    sqs match {
+      case Some(clientAndName) =>
+        if (sqsQueueExists(clientAndName.sqsClient, clientAndName.sqsBufferName)) ()
+        else log.warn(s"SQS queue ${clientAndName.sqsBufferName} is defined in config file, but does not exist.")
```

Comment:
Even if SQS queue is defined, I think that we want to check the existence of Kinesis stream and log a warning if it doesn't exist.

Reply:
Good point, I've added that log message.